portal-impl/src/com/liferay/portal/security/permission/StagingPermissionChecker.java

```diff
@@ -32,6 +36,37 @@
  */
 public class StagingPermissionChecker implements PermissionChecker {
 
+	public static Long getGroupId() {
+		Long groupId = _groupId.get();
+
+		if (_log.isDebugEnabled()) {
+			_log.debug("getGroupId " + groupId);
+		}
+
+		return groupId;
+	}
+
+	public static void removeGroupId() {
+		if (_log.isDebugEnabled()) {
+			_log.debug("removeGroupId ");
+		}
+
+		_groupId.remove();
+	}
+
+	public static void setGroupId(Long groupId) {
+		if (_log.isDebugEnabled()) {
+			_log.debug("setGroupId " + groupId);
+		}
+
+		if (groupId > 0) {
```

Comment:
Why is this check here?

Reply:
yeah, I think we can remove it.

@csierra, is it ok?

Reply:
I think so... I just quickly copied it over from `GroupThreadLocal`

Reply:
Let's just keep it simple and leave out the logging and less than zero checking.  No reason to copy GroupThreadLocal since this new use case is very clear.